samples/bluetooth/hci_vs_scan_req/src/main.c

```diff
@@ -0,0 +1,228 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2024 Giancarlo Stasi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/hci_vs.h>
+#include <zephyr/bluetooth/addr.h>
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LENGTH (sizeof(DEVICE_NAME) - 1)
+
+/* Advertising Interval: the longer, the less energy consumption.
+ * Units: 0.625 milliseconds.
+ * The Minimum Advertising Interval and Maximum Advertising Interval should not be the same value
+ * (as stated in Bluetooth Core Spec 5.2, section 7.8.5)
+ */
+#define ADV_MIN_INTERVAL        BT_GAP_ADV_SLOW_INT_MIN
+#define ADV_MAX_INTERVAL        BT_GAP_ADV_SLOW_INT_MAX
+
+#if defined(CONFIG_BT_PERIPHERAL)
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_SCANNABLE | \
+				 BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#else
+#define ADV_OPTIONS             (BT_LE_ADV_OPT_SCANNABLE | BT_LE_ADV_OPT_NOTIFY_SCAN_REQ)
+#endif
+
+/*
+ * Radio output power for advertisements. Lower the value, closer the scanner.
+ * Values are defined in HAL (e.g. modules/hal/nordic/nrfx/mdk/nrf52_bitfields.h)
+ * signed 8-bit representing dBm.
+ */
+#define ADV_TX_POWER ((int8_t)(-8))
+
+static uint8_t scan_data[] = {'V', 'S', ' ', 'S', 'a', 'm', 'p', 'l', 'e'};
+
+static const struct bt_le_adv_param parameters = {
+	.options = ADV_OPTIONS,
+	.interval_min = ADV_MIN_INTERVAL,
+	.interval_max = ADV_MAX_INTERVAL,
+};
+
+static const struct bt_data adv_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LENGTH),
+};
+
+static const struct bt_data scan_rsp_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, scan_data, sizeof(scan_data)),
+};
+
+static const char *bt_addr_le_str(const bt_addr_le_t *addr)
+{
+	static char str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, str, sizeof(str));
+
+	return str;
+}
+
+/* Allow different TX power per role/connection. It requires CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL.
+ * same as in samples/bluetooth/hci_pwr_ctrl/src
+ */
+static void set_tx_power(uint8_t handle_type, uint16_t handle, int8_t tx_pwr_lvl)
```

Comment:
Does this sample really need to set the TX power?
I think you should remove it if not absolutely necessary, as to make the sample as focused and streamlined as possible.

Reply:
It's necessary in our app module, not in this sample: removed.